src/core_plugins/kibana/public/dashboard/panel/panel.js

```diff
@@ -117,6 +118,20 @@ uiModules
           $scope.$watchCollection('panel.sort', function () {
             $scope.saveState();
           });
+
+          $scope.addColumn = function addColumn(columnName) {
+            $scope.savedObj.searchSource.get('index').popularizeField(columnName, 1);
+            columnActions.addColumn($scope.panel.columns, columnName);
+          };
+
+          $scope.removeColumn = function removeColumn(columnName) {
+            $scope.savedObj.searchSource.get('index').popularizeField(columnName, 1);
```

Comment:
You're right, this was previously not consistently triggered for the different ways to add columns. I actually didn't notice, but I guess we could keep it.